### PR TITLE
Do not throw in ClassLoaderTable

### PR DESCRIPTION
**Pull request by cjjdespres**
In `copyJ9UTF8()`, we now avoid throwing `std::bad_alloc()` if memory for a copy of a ROM class's name could not be allocated. We instead do not store the `TR_ClassLoaderInfo` for that class in the `_nameTable` in such situations. This parallels the treatment of the class loader info if a chain for the loader could not be found in the SCC.

### CODE REVIEW

**Comment by cjjdespres** [comment]
Attn @mpirvu. Part of https://github.com/eclipse-openj9/openj9/issues/18990.

File: runtime/compiler/env/ClassLoaderTable.cpp [hunk 1]
```diff
@@ -293,7 +295,7 @@ TR_PersistentClassLoaderTable::associateClassLoaderWithClass(J9VMThread *vmThrea
       }
 
 #if defined(J9VM_OPT_JITSERVER)
-   if (useAOTCache)
+   if (nameStr)
```
[the remark this answers is not in the document]
Comment (cjjdespres) [reply runtime/compiler/env/ClassLoaderTable.cpp:298]: That is true. I don't think it's harmful, exactly, but we could avoid adding a useless entry if both chain and name retrieval failed. I'll change that.

**Comment by cjjdespres** [comment]
Force-pushed to change the early return logic in `associateClassLoaderWithClass()` to avoid adding a useless entry.

**Review by mpirvu — APPROVED** [review]
LGTM

**Comment by mpirvu** [comment]
jenkins test sanity plinuxjit,xlinuxjit,zlinuxjit,alinux64jit jdk17

**Comment by mpirvu** [comment]
jenkins compile win jdk17

**Pull request merged**